### PR TITLE
Make AddComponentReferenceCapture delegate non-null

### LINKED ISSUE #24794: Capture reference to blazor components with nullability enabled always gives warning on net5.0 preview 7
When project nullability is enabled there is no way to capture a reference to a blazor component without generating a compiler warning.

### Steps to reproduce
* Create template project: `dotnet new blazorwasm`
* Edit project file, add `<nullable>enable</nullable>`
* Build -> observe there are some nullable warnings, ignore these they can all be fixed by updating the code
* In `Index.razor` add the following:
```
@code {
   SurveyPrompt p;

}
```
* In `Index.razor` update the component line to read: `<SurveyPrompt @ref=p Title="How is Blazor working for you?" />`
* Build the project again -> observe warning line #7 `CS8618: Non-nullable field 'p' is uninitialized'`
* Change declaration to: `SurveyPrompt? p;`
* Build again -> observe warning line #7`CS8600: Converting null literal or possible null value to non-nullable type`

<b>There seems to be no way to get rid of this warning. If it was somewhere in c# code one could have added a pragma around the specific line, but since the warning is in the razor part of the file it cannot be disabled for that specific line only</b>

### PR DESCRIPTION
**Pull request by pranavkm**
* Make the sample nullable enabled

Fixes https://github.com/dotnet/aspnetcore/issues/24794

### CODE REVIEW

File: src/Components/Samples/BlazorServerApp/Data/WeatherForecast.cs [hunk 1]
```diff
@@ -10,6 +10,6 @@ public class WeatherForecast
 
         public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);
 
-        public string Summary { get; set; }
+        public string Summary { get; set; } = default!;
```
Comment (Youssef1313) [inline src/Components/Samples/BlazorServerApp/Data/WeatherForecast.cs:13]: Why not make it a nullable string (`string?`)?

**Review by SteveSandersonMS — APPROVED** [review]
Doesn’t seem like the js files should have been changed. Approving apart from that though!

**Comment by pranavkm** [comment]
@SteveSandersonMS the JS files in master appear to out of date.

File: src/Components/Samples/BlazorServerApp/Data/WeatherForecast.cs (hunk 1 above)
Comment (pranavkm) [reply src/Components/Samples/BlazorServerApp/Data/WeatherForecast.cs:13]: In this particular case, Summary is always initialized, so this better reflects the nullability of the property.

Comment (Youssef1313) [reply src/Components/Samples/BlazorServerApp/Data/WeatherForecast.cs:13]: @pranavkm If I do `var forecast = new WeatherForecast();`, it won't be initialized. Giving it a default value of `string.Empty` could be better IMO.

**Pull request merged**